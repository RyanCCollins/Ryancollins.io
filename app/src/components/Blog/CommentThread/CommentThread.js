import React from 'react';
import ReactDisqusThread from 'react-disqus-thread'
import {
  Row,
  Column
} from 'react-foundation';

const CommentThread = ({
  post,
  onNewComment,
  location
}) => (
  <Row className="comment-thread-container">
    <Column isColumn small={12} large={8} centerOnSmall>
      <ReactDisqusThread
        shortname="ryancollinsio"
        identifier={`ryancollinsio-${post._id}`}
        title={post.title}
        url={`https://ryancollins.io${location.pathname}`}
        onNewComment={onNewComment}
      />
    </Column>
  </Row>
);

CommentThread.propTypes = {
  post: React.PropTypes.object.isRequired,
  onNewComment: React.PropTypes.func.isRequired,
  location: React.PropTypes.object.isRequired
};

export default CommentThread;
