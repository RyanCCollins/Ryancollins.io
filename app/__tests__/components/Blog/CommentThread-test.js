import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import CommentThread from '../../../src/components/Blog/CommentThread/CommentThread';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('<CommentThread />', () => {
  const mockedPost = {
    title: 'Hello World',
  };
  const mockedLocation = {
    pathname: '/blog'
  };
  it('renders with required props', () => {
    const component = renderer.create(
      <CommentThread
        post={mockedPost}
        location={mockedLocation}
        onNewComment={(e) => e}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
