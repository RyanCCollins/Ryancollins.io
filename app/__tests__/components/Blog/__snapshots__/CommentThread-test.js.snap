exports[`<CommentThread /> renders with required props 1`] = `
<div
  className="row comment-thread-container">
  <div
    centerOnSmall={true}
    className="column small-12 large-8 small-centered"
    isColumn={true}
    large={8}
    small={12}>
    <div
      category_id={null}
      identifier="ryancollinsio-undefined"
      onNewComment={[Function onNewComment]}
      shortname="ryancollinsio"
      title="Hello World"
      url="https://ryancollins.io/blog">
      <div
        id="disqus_thread" />
      <noscript>
        <span>
          Please enable JavaScript to view the
          <a
            href="http://disqus.com/?ref_noscript">
            comments powered by Disqus.
          </a>
        </span>
      </noscript>
      <a
        className="dsq-brlink"
        href="http://disqus.com">
        Blog comments powered by 
        <span
          className="logo-disqus">
          Disqus
        </span>
        .
      </a>
    </div>
  </div>
</div>
`;
